cg/meta/workflow/analysis.py

```diff
@@ -401,27 +407,18 @@ def decompress_case(self, case_id: str, dry_run: bool) -> None:
             self.set_statusdb_action(case_id=case_id, action="analyze")
         LOG.info("Decompression started for %s", case_id)
 
-    def resolve_decompression(self, case_id: str, dry_run: bool) -> bool:
+    def resolve_decompression(self, case_id: str, dry_run: bool) -> None:
         """
-        Handles decompression automatically for case.
-        Return True if decompression is running
+        Decompresses a case if needed and adds new fastq files to Housekeeper.
         """
         if self.prepare_fastq_api.is_spring_decompression_needed(case_id):
             self.decompress_case(case_id=case_id, dry_run=dry_run)
-            return True
+            return
 
         if self.prepare_fastq_api.is_spring_decompression_running(case_id):
```

Comment:
Side track, should not this check be made first? I guess `is_spring_decompression_needed` also checks whether it is running already so might not matter

Reply:
`is_spring_decompression_needed` is not checking if decompression is running. Swapping the order would also mean that if some files are being decompressed, we would miss decompressing any files which might need to be decompressed but which are not currently being decompressed.

But maybe you meant something like 

```
if decompression running or needed: 
    if needed: 
        decompress 
    return
```

Reply:
Alright, great!